costs/anomalies/anomalies_route.go

```diff
@@ -81,11 +82,21 @@ func makeElasticSearchRequest(ctx context.Context, parsedParams AnomalyEsQueryPa
 	res, err := searchService.Do(ctx)
 	if err != nil {
 		if elastic.IsNotFound(err) {
-			l.Warning("Query execution failed, ES index does not exists : "+index, err)
-			return nil, http.StatusOK, err
+			l.Warning("Query execution failed, ES index does not exists", map[string]interface{}{
+				"index": index,
+				"error": err.Error(),
+			})
+			return nil, http.StatusOK, errors.GetErrorMessage(ctx, err)
+		} else if err.(*elastic.Error).Details.Type == "search_phase_execution_exception" {
+			l.Error("Error while getting data from ES", map[string]interface{}{
+				"type": fmt.Sprintf("%T", err),
+				"error": err,
+			})
+			return nil, http.StatusInternalServerError, errors.GetErrorMessage(ctx, err)
+		} else {
+			l.Error("Query execution failed : "+err.Error(), nil)
+			return nil, http.StatusInternalServerError, fmt.Errorf("could not execute the ElasticSearch query")
```

Comment:
Are you sure you want tell the user such an accurate error? ElasticSearch should not appear to me.